HHS/simpler-grants-gov
# [Issue 2583] opportunity detail error page

## Summary
Fixes #2583

### Time to review: __15 mins__

## Changes proposed
Added a generic error page for the opportunity detail route and adjusted logic to display it in non 404 error situations.

## Context for reviewers

### Test steps

1. visit an opportunity detail page for an opportunity that doesn't exist, ex: http://localhost:3000/es/opportunity/44444444
2. _VERIFY_: the 404 / not found page is displayed
3. update the API such that it sends a 500 error. Ex. update src/api/opportunities_v1/opportunity_routes.py line 198:
```
opportunity = get_opportunity(db_session, "show me an error")
```
4. restart backend containers. Run `make stop && make build && make start` in the API directory
5. visit an opportunity detail page for an opportunity that doesn't exist, ex: http://localhost:3000/es/opportunity/4
6. _VERIFY_: a generic error page is shown

## Additional information

<img width="1331" alt="Screenshot 2024-10-25 at 2 25 10 PM" src="https://github.com/user-attachments/assets/308f8f39-f3b2-453c-a7b6-c85a32fc01bd">
